### PR TITLE
Move and movel

### PR DESCRIPTION
**Pull request by lanarimarco**
MOVEL implementation.
Fix MOVE - Now is allowed to "move" string literal in a target variable

### CODE REVIEW

File: rpgJavaInterpreter-core/src/main/kotlin/com/smeup/rpgparser/interpreter/values.kt [hunk 1]
```diff
@@ -121,7 +128,7 @@ data class IntValue(val value: Long) : NumberValue, Value() {
 
     override fun asInt() = this
     // TODO Verify conversion
-    override fun asDecimal(): DecimalValue = DecimalValue(BigDecimal(value))
+    override fun asDecimal(): DecimalValue = DecimalValue(internalValue)
```
Comment (ftomassetti) [inline rpgJavaInterpreter-core/src/main/kotlin/com/smeup/rpgparser/interpreter/values.kt]: I would add a blank line before the comment. Also, I do not undersand what needs to be verified here.
Finally, in this case you can remove the return type of the method as it is obvious

File: rpgJavaInterpreter-core/src/main/kotlin/com/smeup/rpgparser/interpreter/values.kt [hunk 2]
```diff
@@ -332,7 +343,7 @@ class ProjectedArrayValue(val container: ArrayValue, val field: FieldDefinition)
             if (value is StringValue) {
                 containerElement.setSubstring(field.startOffset, field.endOffset, value)
             } else if (value is IntValue) {
-                var s = value.value.toString()
+                val s = value.value.toString()
```
Comment (ftomassetti) [inline rpgJavaInterpreter-core/src/main/kotlin/com/smeup/rpgparser/interpreter/values.kt]: Good

File: rpgJavaInterpreter-core/src/main/kotlin/com/smeup/rpgparser/parsing/parsetreetoast/misc.kt [hunk 3]
```diff
@@ -155,16 +155,24 @@ fun literalToNumber(
     text: String,
     position: Position?
 ): Expression {
+    // fix minus at right
```
Comment (ftomassetti) [inline rpgJavaInterpreter-core/src/main/kotlin/com/smeup/rpgparser/parsing/parsetreetoast/misc.kt]: This is not clear to me. What is the goal of this? Could you rephrase the comment please?

File: mute_tests/MUTE13_04.rpgle [hunk 4]
```diff
@@ -1,70 +1,89 @@
...
-     C                   DSPLY                   DSP
-    MU* VAL1(AAA010) VAL2('ABCDEWXYZ ') COMP(EQ)
-     C                   EVAL      AAA010='RSTUVWXYZ'
-     C                   EVAL      AAA005='ABCDE'
-     C                   MOVEL     AAA005        AAA010
-     C                   EVAL      DSP=AAA010
-     C                   DSPLY                   DSP
-      *
-     C                   SETON                                        LR
+   COP* *NOUI
```
[the remark this answers is not in the document]
Comment (lanarimarco) [reply mute_tests/MUTE13_04.rpgle]: I'll try

File: rpgJavaInterpreter-core/src/main/kotlin/com/smeup/rpgparser/interpreter/movel.kt [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+import com.smeup.rpgparser.parsing.ast.Expression
+import java.math.BigDecimal
+
+private fun assignStringToString(operationExtender: String?, target: DataRefExpr, factor2: Expression, interpreterCoreHelper: InterpreterCoreHelper): Value {
+    var newValue = interpreterCoreHelper.interpret(factor2)
+    if (factor2.type().size > target.size()) {
+        newValue = newValue.takeFirst(target.size().toInt())
+    } else if (factor2.type().size < target.size()) {
+        val append = if (operationExtender == null) {
+            interpreterCoreHelper.get(target.variable.referred!!).takeLast(target.size().toInt() - factor2.type().size.toInt())
```
[the remark this answers is not in the document]
Comment (lanarimarco) [reply rpgJavaInterpreter-core/src/main/kotlin/com/smeup/rpgparser/interpreter/movel.kt]: Ok

File: mute_tests/MUTE13_04.rpgle [hunk 6]
```diff
@@ -1,70 +1,89 @@
...
+     C                   MOVEL(P)  AAA005        AAA010
+     C                   EVAL      DSP=AAA010
+     C                   DSPLY                   DSP
+     C                   EVAL      AAA010='RSTUVWXYZ'
+     C                   EVAL      AAA005='ABCDE'
+    MU* VAL1(AAA010) VAL2('ABCDEWXYZ ') COMP(EQ)
+     C                   MOVEL     AAA005        AAA010
+     C                   EVAL      DSP=AAA010
+     C                   DSPLY                   DSP
+      *    Facendo la movel in una stringa di un numero negativo l'ultima cifra diventa una lettera :
```
[the remark this answers is not in the document]
Comment (lanarimarco) [reply mute_tests/MUTE13_04.rpgle]: I will delete this MUTE.

File: rpgJavaInterpreter-core/src/main/kotlin/com/smeup/rpgparser/interpreter/values.kt [hunk 7]
```diff
@@ -112,6 +113,12 @@ fun String.removeNullChars(): String {
 }
 
 data class IntValue(val value: Long) : NumberValue, Value() {
+
+    private val internalValue = BigDecimal(value)
+
+    override val bigDecimal: BigDecimal
+        get() = BigDecimal(value)
```
[the remark this answers is not in the document]
Comment (lanarimarco) [reply rpgJavaInterpreter-core/src/main/kotlin/com/smeup/rpgparser/interpreter/values.kt]: Ops!!! Wrong choice

File: rpgJavaInterpreter-core/src/main/kotlin/com/smeup/rpgparser/interpreter/movel.kt [hunk 8]
```diff
@@ -0,0 +1,58 @@
+package com.smeup.rpgparser.interpreter
+
+import com.smeup.rpgparser.parsing.ast.AssignableExpression
+import com.smeup.rpgparser.parsing.ast.DataRefExpr
+import com.smeup.rpgparser.parsing.ast.Expression
+import java.math.BigDecimal
+
+private fun assignStringToString(operationExtender: String?, target: DataRefExpr, factor2: Expression, interpreterCoreHelper: InterpreterCoreHelper): Value {
```
[the remark this answers is not in the document]
Comment (ftomassetti) [reply rpgJavaInterpreter-core/src/main/kotlin/com/smeup/rpgparser/interpreter/movel.kt]: My question is: would every contributor to this project, present and future, know what operationExtender and factor2 means? If so, good, otherwise we should add a comment

File: rpgJavaInterpreter-core/src/main/kotlin/com/smeup/rpgparser/parsing/parsetreetoast/misc.kt (hunk 3 above)
Comment (ftomassetti) [reply rpgJavaInterpreter-core/src/main/kotlin/com/smeup/rpgparser/parsing/parsetreetoast/misc.kt]: This would be a nice example to add as a comment. Also, is it ok that "2-" is interpreted as the number 2 or should we produce an error?

File: rpgJavaInterpreter-core/src/main/kotlin/com/smeup/rpgparser/interpreter/movel.kt [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+        newValue = newValue.takeFirst(target.size().toInt())
+    } else if (factor2.type().size < target.size()) {
+        newValue = newValue.concatenate(interpreterCoreHelper.get(target.variable.referred!!).takeLast(target.size().toInt() - factor2.type().size.toInt()))
+    }
+    return interpreterCoreHelper.assign(target, newValue)
+}
+
+fun movel(operationExtender: String?, target: AssignableExpression, value: Expression, interpreterCoreHelper: InterpreterCoreHelper): Value {
+    require(target is DataRefExpr)
+    return if (value.type() is StringType && target.type() is StringType) {
```
[the remark this answers is not in the document]
Comment (ftomassetti) [reply rpgJavaInterpreter-core/src/main/kotlin/com/smeup/rpgparser/interpreter/movel.kt]: If there is a specific behavior your choice is the correct one. I will close this comment.

File: rpgJavaInterpreter-core/src/main/kotlin/com/smeup/rpgparser/parsing/parsetreetoast/misc.kt (hunk 3 above)
Comment (lanarimarco) [reply rpgJavaInterpreter-core/src/main/kotlin/com/smeup/rpgparser/parsing/parsetreetoast/misc.kt]: "2-" has to be interpreted as -2
